open-hackathon-server/src/hackathon/hack/hackathon_manager.py

```diff
@@ -175,10 +175,8 @@ def get_user_hackathon_list_with_detail(self, user_id):
         return map(lambda h: self.__get_hackathon_detail(h, user), user_hack_list)
 
     def get_recyclable_hackathon_list(self):
-        # todo fix auto recycle
-        # all_hackathon = self.db.find_all_objects(Hackathon)
-        # return filter(lambda h: self.is_recycle_enabled(h), all_hackathon)
-        return []
+        hackathons = Hackathon.objects().all()
+        return filter(lambda h: self.is_recycle_enabled(h), hackathons)
```

Comment:
Please add `todo` item here. We don't need read all hackathons and then filter them. It's better to filter on db level.